cluster/juju/layers/kubernetes-worker/templates/ingress-daemon-set.yaml

```diff
@@ -94,86 +151,127 @@ rules:
-kind: ConfigMap
-metadata:
-  name: nginx-load-balancer-{{ juju_application }}-conf
+    name: nginx-ingress-serviceaccount-{{ juju_application }}
+    namespace: ingress-nginx-{{ juju_application }}
+
 ---
 apiVersion: {{ daemonset_api_version }}
 kind: DaemonSet
 metadata:
-  name: nginx-ingress-{{ juju_application }}-controller
+  name: nginx-ingress-controller-{{ juju_application }}
+  namespace: ingress-nginx-{{ juju_application }}
   labels:
+    app.kubernetes.io/name: ingress-nginx-{{ juju_application }}
+    app.kubernetes.io/part-of: ingress-nginx-{{ juju_application }}
     juju-application: nginx-ingress-{{ juju_application }}
+    cdk-{{ juju_application }}-ingress: "true"
 spec:
   selector:
     matchLabels:
-      name: nginx-ingress-{{ juju_application }}
+      app.kubernetes.io/name: ingress-nginx-{{ juju_application }}
+      app.kubernetes.io/part-of: ingress-nginx-{{ juju_application }}
   template:
     metadata:
       labels:
-        name: nginx-ingress-{{ juju_application }}
+        app.kubernetes.io/name: ingress-nginx-{{ juju_application }}
+        app.kubernetes.io/part-of: ingress-nginx-{{ juju_application }}
+      annotations:
+        prometheus.io/port: "10254"
+        prometheus.io/scrape: "true"
     spec:
+      serviceAccountName: nginx-ingress-serviceaccount-{{ juju_application }}
       nodeSelector:
         juju-application: {{ juju_application }}
       terminationGracePeriodSeconds: 60
       # hostPort doesn't work with CNI, so we have to use hostNetwork instead
       # see https://github.com/kubernetes/kubernetes/issues/23920
       hostNetwork: true
-      serviceAccountName: nginx-ingress-{{ juju_application }}-serviceaccount
       containers:
-      - image: {{ ingress_image }}
-        name: nginx-ingress-{{ juju_application }}
-        livenessProbe:
-          httpGet:
-            path: /healthz
-            port: 10254
-            scheme: HTTP
-          initialDelaySeconds: 30
-          timeoutSeconds: 5
-        # use downward API
-        env:
-          - name: POD_NAME
-            valueFrom:
-              fieldRef:
-                fieldPath: metadata.name
-          - name: POD_NAMESPACE
-            valueFrom:
-              fieldRef:
-                fieldPath: metadata.namespace
-        ports:
-        - containerPort: 80
-        - containerPort: 443
-        args:
-        - /nginx-ingress-controller
-        - --default-backend-service=$(POD_NAMESPACE)/default-http-backend
-        - --configmap=$(POD_NAMESPACE)/nginx-load-balancer-conf
-        - --enable-ssl-chain-completion={{ ssl_chain_completion }}
+        - name: nginx-ingress-controller{{ juju_application }}
+          image: {{ ingress_image }}
+          args:
+            - /nginx-ingress-controller
+            - --configmap=$(POD_NAMESPACE)/nginx-configuration
+            - --tcp-services-configmap=$(POD_NAMESPACE)/tcp-services
+            - --udp-services-configmap=$(POD_NAMESPACE)/udp-services
```

Comment:
The values for --configmap, --tcp-services-configmap, and --udp-services-configmap are all missing the `-{{ juju_application }}` suffix.